src/auth/handlers.rs

```diff
@@ -25,15 +24,26 @@ use time::OffsetDateTime;
 async fn register(
     ctx: Data<Context>,
     Json(user_data): Json<SignUpCredentials>,
-) -> Result<HttpResponse> {
+) -> HttpResponse {
     log::trace!("Received register request");
 
     let mut user = user_data;
-    user.password = bcrypt::hash(user.password, bcrypt::DEFAULT_COST)?;
-
-    ctx.db.add_user(user).await?;
+    user.password = match bcrypt::hash(user.password, bcrypt::DEFAULT_COST) {
+        Ok(hashed_password) => hashed_password,
+        Err(_) => {
+            return HttpResponse::BadRequest().body("Hashing password error");
```

Comment:
В функции `login` ошибка `bcrypt::hash` интерпретируется как `InternalServerError`, так что надо выбрать что-то одно. Считаю, что это должно быть ISE, потому что фильтрацией ввода должен заниматься не бэкенд.

```suggestion
            return HttpResponse::InternalServerError().body("Hashing password error");
```